### PR TITLE
Initial support for windows and mac

### PR DESCRIPTION
**Pull request by tbouffard**
Previous testings has shown that the existing script allows to build on Windows at the price of commenting some parts only required for Linux.
With this PR, I propose to let the script detect the OS and skip the Linux specific actions to make it work out of the box on other OS.

- [x] windows tests ✅
  - git bash `2.20.0.windows.1` --> `MINGW64_NT-10.0  2.11.2(0.329/5/3) 2018-11-10 14:38 x86_64 Msys`
  - Windows `10 Pro 1903 18362.418`
  - Java `Java version: 1.8.0_212, vendor: AdoptOpenJDK`
  - Default locale: `fr_FR, platform encoding: Cp1252`
```
Build is running on Windows/MINGW
> Using Maven version: 3.6.0
> Using curl version: 7.62.0
Check if Java version is compatible with Bonita
> JAVA_HOME is set
> Java command path is C:\Program Files\Java\jdk8u212-b04/bin/java
> Java full version: 1.8.0_212
> Java version: 8
```
- [ ] mac tests
- [x] linux tests (waiting for our Travis CI friend 😺 )

### CODE REVIEW

**Comment by tbouffard** [comment]
@laurentleseigneur if you want/can give it a try on Mac, I will be pleased to get your feedbacks
Thanks

File: build-script.sh [hunk 1]
```diff
@@ -357,6 +367,7 @@ detectWebPagesDependenciesVersions() {
 # MAIN
 ########################################################################################################################
 checkPrerequisites
+exit 12
```
[the remark this answers is not in the document]
Comment (tbouffard) [reply build-script.sh]: exactly, should be removed 😈

**Comment by laurentleseigneur** [comment]
yes i can test on Mac

**Comment by tbouffard** [comment]
@amottier I have preliminary tests to do on Windows first, I will ping you when I am done

File: build-script.sh (hunk 1 above)
Comment (tbouffard) [inline build-script.sh]:
```suggestion
```

**Comment by amottier** [comment]
@tbouffard @dcoille successfully run the script on macOS.

**Comment by laurentleseigneur** [comment]
fails on mac os for my part
`tr` has wrong arguments

```
Detecting web-pages dependencies versions
tr: illegal option -- -
usage: tr [-Ccsu] string1 string2
       tr [-Ccu] -d string1
       tr [-Ccu] -s string1
       tr [-Ccu] -ds string1 string2
tr: illegal option -- -
usage: tr [-Ccsu] string1 string2
       tr [-Ccu] -d string1
       tr [-Ccu] -s string1
       tr [-Ccu] -ds string1 string2
WEB_PAGES_UID_VERSION:
============================================================
Processing bonita-ui-designer 7.9.4
============================================================
remote: Enumerating objects: 1447, done.
remote: Counting objects: 100% (1319/1319), done.
remote: Compressing objects: 100% (381/381), done.
remote: Total 1021 (delta 663), reused 907 (delta 552), pack-reused 0
Receiving objects: 100% (1021/1021), 178.44 KiB | 842.00 KiB/s, done.
Resolving deltas: 100% (663/663), completed with 181 local objects.
From https://github.com/bonitasoft/bonita-ui-designer
 * [new tag]           1.10.14    -> 1.10.14
 * [new tag]           1.10.15    -> 1.10.15
 * [new tag]           1.10.16    -> 1.10.16
 * [new tag]           1.10.17    -> 1.10.17
 * [new tag]           1.10.18    -> 1.10.18
 * [new tag]           1.10.19    -> 1.10.19
 * [new tag]           1.10.20    -> 1.10.20
 * [new tag]           1.10.21    -> 1.10.21
 * [new tag]           1.10.22    -> 1.10.22
 * [new tag]           1.10.23    -> 1.10.23
 * [new tag]           1.10.24    -> 1.10.24
 * [new tag]           1.9.62     -> 1.9.62
 * [new tag]           1.9.63     -> 1.9.63
 * [new tag]           1.9.64     -> 1.9.64
 * [new tag]           1.9.65     -> 1.9.65
fatal: ambiguous argument 'tags/7.9.4': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
╭─laurent@Bonita ~/git/Build-Bonita ‹initial_support_for_windows_and_mac›
```

**Comment by laurentleseigneur** [comment]
also fails with studio at sed step:

```
============================================================
Processing bonita-studio 7.9.4
============================================================
Cloning into 'bonita-studio'...
remote: Enumerating objects: 781, done.
remote: Counting objects: 100% (781/781), done.
remote: Compressing objects: 100% (402/402), done.
remote: Total 495111 (delta 257), reused 587 (delta 172), pack-reused 494330
Receiving objects: 100% (495111/495111), 368.90 MiB | 2.17 MiB/s, done.
Resolving deltas: 100% (353553/353553), done.
Checking out files: 100% (8332/8332), done.
HEAD is now at a162d70322 [STUDIO RELEASE] Prepare release 7.9.4
WARN: workaround on bonita-studio - fix platform.target URL
sed: 1: "platform/platform.target": extra characters at the end of p command
```

**Comment by tbouffard** [comment]
@laurentleseigneur @amottier it seems like the Mac tr and sed tools does not work as on Linux/Windows :smiling_imp: 
Could you provide the version of these tools installed on your Mac?

**Comment by amottier** [comment]
@tbouffard I think I found a page that provide the solution for the sed part: https://gradot.wordpress.com/2013/12/08/sed-i-sous-mac-os-x/
Do you deal with update of `tr` and `sed` parameter or do you want me do it?

@laurentleseigneur it seems that https://ss64.com/osx/sed.html is not an up to date version of the man page of sed for macOS as it does not mention the `-i` option whereas it seems available on macOS.

**Comment by tbouffard** [comment]
@amottier I don't have MacOS environment, so you can manage this (notice that @laurentleseigneur already managed the `tr` issue)

**Comment by dcoille** [comment]
@tbouffard I tested on MacOS and the build failed :

`Unzipping /Users/delphine.bonitasoft/.gradle/wrapper/dists/gradle-4.5.1-all/87kuriyahurjjkki3zii366f2/gradle-4.5.1-all.zip to /Users/delphine.bonitasoft/.gradle/wrapper/dists/gradle-4.5.1-all/87kuriyahurjjkki3zii366f2
Set executable permissions for: /Users/delphine.bonitasoft/.gradle/wrapper/dists/gradle-4.5.1-all/87kuriyahurjjkki3zii366f2/gradle-4.5.1/bin/gradle
Starting a Gradle Daemon (subsequent builds will be faster)
Generating JAR file 'gradle-api-4.5.1.jar'
Download https://plugins.gradle.org/m2/com/moowork/gradle/gradle-node-plugin/1.2.0/gradle-node-plugin-1.2.0.pom
Download https://plugins.gradle.org/m2/com/moowork/gradle/gradle-node-plugin/1.2.0/gradle-node-plugin-1.2.0.jar
Download https://plugins.gradle.org/m2/com/bonitasoft/gradle/bonita-release/com.bonitasoft.gradle.bonita-release.gradle.plugin/0.1.43/com.bonitasoft.gradle.bonita-release.gradle.plugin-0.1.43.pom
Download https://plugins.gradle.org/m2/com/bonitasoft/gradle/bonita-gradle-plugins/0.1.43/bonita-gradle-plugins-0.1.43.pom

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'bonita-web-pages'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve com.jcraft:jsch.agentproxy.core:0.0.9.
     Required by:
         project : > com.bonitasoft.gradle.bonita-release:com.bonitasoft.gradle.bonita-release.gradle.plugin:0.1.43 > com.bonitasoft.gradle:bonita-gradle-plugins:0.1.43 > org.ajoberstar:grgit:2.2.1
         project : > com.bonitasoft.gradle.bonita-release:com.bonitasoft.gradle.bonita-release.gradle.plugin:0.1.43 > com.bonitasoft.gradle:bonita-gradle-plugins:0.1.43 > org.ajoberstar:grgit:2.2.1 > com.jcraft:jsch.agentproxy.jsch:0.0.9
         project : > com.bonitasoft.gradle.bonita-release:com.bonitasoft.gradle.bonita-release.gradle.plugin:0.1.43 > com.bonitasoft.gradle:bonita-gradle-plugins:0.1.43 > org.ajoberstar:grgit:2.2.1 > com.jcraft:jsch.agentproxy.pageant:0.0.9
         project : > com.bonitasoft.gradle.bonita-release:com.bonitasoft.gradle.bonita-release.gradle.plugin:0.1.43 > com.bonitasoft.gradle:bonita-gradle-plugins:0.1.43 > org.ajoberstar:grgit:2.2.1 > com.jcraft:jsch.agentproxy.sshagent:0.0.9
         project : > com.bonitasoft.gradle.bonita-release:com.bonitasoft.gradle.bonita-release.gradle.plugin:0.1.43 > com.bonitasoft.gradle:bonita-gradle-plugins:0.1.43 > org.ajoberstar:grgit:2.2.1 > com.jcraft:jsch.agentproxy.usocket-jna:0.0.9
         project : > com.bonitasoft.gradle.bonita-release:com.bonitasoft.gradle.bonita-release.gradle.plugin:0.1.43 > com.bonitasoft.gradle:bonita-gradle-plugins:0.1.43 > org.ajoberstar:grgit:2.2.1 > com.jcraft:jsch.agentproxy.usocket-nc:0.0.9
      > Could not resolve com.jcraft:jsch.agentproxy.core:0.0.9.
         > Could not get resource 'http://repositories.rd.lan/maven/all/com/jcraft/jsch.agentproxy.core/0.0.9/jsch.agentproxy.core-0.0.9.pom'.
            > Could not HEAD 'http://repositories.rd.lan/maven/all/com/jcraft/jsch.agentproxy.core/0.0.9/jsch.agentproxy.core-0.0.9.pom'.
               > Connect to repositories.rd.lan:80 [repositories.rd.lan/192.168.4.6] failed: connect timed out

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 5m 9s`

**Comment by amottier** [comment]
@dcoille I open an issue (BPO-301) in our internal bug tracker for bonita-web-pages to remove the reference to Bonita server (repositories.rd.lan) that his only available from Bonitasoft network and make the build fail as you experienced.

**Comment by laurentleseigneur** [comment]
build is falling at this part for me:

```
[ERROR] Internal error: java.lang.RuntimeException: Could not resolve target platform specification artifact org.bonitasoft.studio:platform:target:7.9.4 -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Could not resolve target platform specification artifact org.bonitasoft.studio:platform:target:7.9.4
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:122)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:55)
Caused by: java.lang.RuntimeException: Could not resolve target platform specification artifact org.bonitasoft.studio:platform:target:7.9.4
    at org.eclipse.tycho.core.resolver.DefaultTargetPlatformConfigurationReader.addTargetArtifact (DefaultTargetPlatformConfigurationReader.java:389)
    at org.eclipse.tycho.core.resolver.DefaultTargetPlatformConfigurationReader.setTarget (DefaultTargetPlatformConfigurationReader.java:342)
    at org.eclipse.tycho.core.resolver.DefaultTargetPlatformConfigurationReader.getTargetPlatformConfiguration (DefaultTargetPlatformConfigurationReader.java:75)
    at org.eclipse.tycho.core.resolver.DefaultTychoResolver.setupProject (DefaultTychoResolver.java:87)
    at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead (TychoMavenLifecycleParticipant.java:90)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:55)
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/InternalErrorException
```

**Comment by dcoille** [comment]
Same for me : 

> [ERROR] Internal error: java.lang.RuntimeException: Could not resolve target platform specification artifact org.bonitasoft.studio:platform:target:7.9.4 -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Could not resolve target platform specification artifact org.bonitasoft.studio:platform:target:7.9.4
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:122)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:55)
Caused by: java.lang.RuntimeException: Could not resolve target platform specification artifact org.bonitasoft.studio:platform:target:7.9.4
    at org.eclipse.tycho.core.resolver.DefaultTargetPlatformConfigurationReader.addTargetArtifact (DefaultTargetPlatformConfigurationReader.java:389)
    at org.eclipse.tycho.core.resolver.DefaultTargetPlatformConfigurationReader.setTarget (DefaultTargetPlatformConfigurationReader.java:342)
    at org.eclipse.tycho.core.resolver.DefaultTargetPlatformConfigurationReader.getTargetPlatformConfiguration (DefaultTargetPlatformConfigurationReader.java:75)
    at org.eclipse.tycho.core.resolver.DefaultTychoResolver.setupProject (DefaultTychoResolver.java:87)
    at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead (TychoMavenLifecycleParticipant.java:90)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:55)
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/InternalErrorException

And I ran the build on my Mac @ home

**Comment by laurentleseigneur** [comment]
@amottier it still fails due to `repositories.rd.lan` required access, which is a private resource. see attached log

**Comment by tbouffard** [comment]
ℹ️ I am working on a poc that setup Github Actions to build on Windows and MacOS
See https://github.com/tbouffard/Build-Bonita/pull/3

**Pull request merged**